portal/src/util/rolesAndGroups.ts

```diff
@@ -0,0 +1,41 @@
+export interface Searchable {
+  id: string;
+  key: string;
+  name?: string | null;
+}
+
+export function searchRolesAndGroups<T extends Searchable>(
```

Comment:
How roles / groups can be searched should be a model logic of roles / groups, therefore I put it in `model/XX.ts`. 

I prefer keeping them as two separated functions `searchGroups` and `searchRoles`.
You can extract some common logics to util, such as the preprocessing of searchKeyword.